core/drivers/pl022_spi.c

```diff
@@ -181,13 +181,21 @@ enum pl022_spi_mode {
 	PL022_SPI_MODE3 = SSPCR0_SPO1 | SSPCR0_SPH1  /* 0xC0 */
 };
 
+/*
+ * @param[inout]	*num_rxpkts	number of max packets to receive
+ */
 static void pl022_txrx8(struct spi_chip *chip, uint8_t *wdat,
 	uint8_t *rdat, size_t num_txpkts, size_t *num_rxpkts)
 {
 	size_t i = 0;
 	size_t j = 0;
 	struct pl022_data *pd = container_of(chip, struct pl022_data, chip);
 
+	if (*num_rxpkts != num_txpkts) {
+		MSG("WARNING: *num_rxpkts != num_txpkts!\n");
+		MSG("WARNING: Is this what you really want?\n");
+	}
+
```

Comment:
I don't like this. Either `*num_rxpkts != num_txpkts` is a valid use case, or it's not. If it's valid, then the caller will probably not want to see that warning => remove this check.

Should you return early when `*num_rxpkts == 0` (to avoid pulsing `cs_gpio_pin`)?